ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountStatsMySqlStore.java

```diff
@@ -233,8 +241,57 @@ public void storeAccountStats(StatsWrapper statsWrapper) throws SQLException {
       }
     }
     batch.flush();
-    storeMetrics.publishTimeMs.update(System.currentTimeMillis() - startTimeMs);
     storeMetrics.batchSize.update(batchSize);
+    storeMetrics.insertAccountStatsTimeMs.update(System.currentTimeMillis() - startTimeMs);
+
+    long deleteStartTimeMs = System.currentTimeMillis();
+    int deleteStatementCounter = 0;
+    // Now delete the rows that appear in previousStats but not in the currentStats
+    for (Map.Entry<String, StatsSnapshot> prevPartitionMapEntry : prevPartitionMap.entrySet()) {
+      String partitionIdKey = prevPartitionMapEntry.getKey();
+      StatsSnapshot prevAccountStatsSnapshot = prevPartitionMapEntry.getValue();
+      StatsSnapshot currAccountStatsSnapshot = currPartitionMap.get(partitionIdKey);
+      int partitionId = Utils.partitionIdFromStatsPartitionKey(partitionIdKey);
+      if (prevAccountStatsSnapshot != null && prevAccountStatsSnapshot.getSubMap() != null && (
+          currAccountStatsSnapshot == null || currAccountStatsSnapshot.getSubMap() == null)) {
+        accountReportsDao.deleteStorageUsageForPartition(clusterName, hostname, partitionId);
```

Comment:
I feel like deleting partition only happens when replica is decommissioned on current node. No need to delete its rows if partition is empty.

Reply:
On second thought, empty partition should be very rare, so I think it's fine to delete it in mysql.

Reply:
empty partition means all the blobs in this partition are deleted (still in retention maybe, but deleted)